src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.FeatureCollection.cs

```diff
@@ -193,6 +190,64 @@ bool IHttpBodyControlFeature.AllowSynchronousIO
             }
         }
 
+        PipeWriter IResponseBodyPipeFeature.ResponseBodyPipe
+        {
+            get
+            {
+                if (_userSetPipeWriter != null)
+                {
+                    return _userSetPipeWriter;
```

Comment:
Where is this getting reset back to null? You should add a regression test similar to RequestTests.StreamsAreNotPersistedAcrossRequests but for Pipes.